File: backend/fuzfriend.ProductsApi/fuzfriend.ProductsApi/Models/Product.cs
using System.ComponentModel.DataAnnotations;

namespace fuzfriend.ProductsApi.Models;

public class Product
{
    public int Id { get; set; }
    [Required]
    public string Title { get; set; } = string.Empty;

    public string Description { get; set; } = string.Empty;

    [Required]
    public string Brand { get; set; } = string.Empty;

    [Required]
    public string Category { get; set; } = string.Empty;

    public string Color { get; set; } = string.Empty;

    public string Size { get; set; } = string.Empty;

    public decimal Price { get; set; }

    public double Rating { get; set; }

    public bool OnPromotion { get; set; }

    public List<string> ImageUrls { get; set; } = new();
}

File: frontend/types/Product.ts
export interface Product {
    id: number;
    title: string;
    description: string;
    imageUrl: string;
    price: number;
    brand: string;
    color: string;
    size: string;
    rating: number;
    onPromotion: boolean;
    category: string;
}

export interface ProductSearchRequest {
    category?: string;
    categories?: string[];
    brands?: string[];
    colours?: string[];
    sizes?: string[];
    minPrice?: number;
    maxPrice?: number;
    minRating?: number;
    onPromotion?: boolean;
    page?: number;
    pageSize?: number;
    sortBy?: string;
    sortDirection?: string;
}

export interface FilterOptions {
    categories: string[];
    brands: string[];
    colours: string[];
    sizes: string[];
    minPrice: number;
    maxPrice: number;
    ratings: number[];
    hasPromotions: boolean;
}

export interface ProductSearchResponse {
    items: Product[];
    totalCount: number;
    filters?: FilterOptions;
}
